hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxyImpl.java

```diff
@@ -393,6 +393,28 @@ public boolean isLocked(@Nonnull K key) {
                 putAsyncInternal(key, valueData, ttl, ttlUnit, maxIdle, maxIdleUnit));
     }
 
+    public InternalCompletableFuture<V> putIfAbsentAsync(@Nonnull K key, @Nonnull V value) {
+        return putIfAbsentAsync(key, value, UNSET, TimeUnit.MILLISECONDS);
+    }
+
+    public InternalCompletableFuture<V> putIfAbsentAsync(@Nonnull K key, @Nonnull V value,
+                                                    long ttl, @Nonnull TimeUnit timeunit) {
+        return putIfAbsentAsync(key, value, ttl, timeunit, UNSET, TimeUnit.MILLISECONDS);
+    }
+
+    public InternalCompletableFuture<V> putIfAbsentAsync(@Nonnull K key, @Nonnull V value,
+                                                         long ttl, @Nonnull TimeUnit timeunit,
+                                                         long maxIdle, @Nonnull TimeUnit maxIdleUnit) {
+        checkNotNull(key, NULL_KEY_IS_NOT_ALLOWED);
+        checkNotNull(value, NULL_VALUE_IS_NOT_ALLOWED);
+        checkNotNull(timeunit, NULL_TIMEUNIT_IS_NOT_ALLOWED);
+
```

Comment:
Minor: could you also add check for `maxIdleUnit` to fail fast? I think it can be useful to validate the exception messages in the tests (if it sounds like overkill, you can ignore this).